libraries/AP_Scripting/examples/Thrust_Stand.lua

```diff
@@ -0,0 +1,1339 @@
+    if cal_status == 0 then
+      gcs:send_text(0,"Inialization done: " .. _dev_name[_nau7802_index])
+      _dev_initialised[_nau7802_index] = true
+
+      -- Reset _nau7802 calibration variables
+      _nau7802_setup_started = false
+
+      -- Device initalised, return to main code
+      return init, 100
+    end
+
+    if cal_status == 1 then
+      -- Cal in progress, wait
+      gcs:send_text(4, "DB: cal in progress")
+      return init_nau7802, 100
+    end
+
+    -- If we got this far then calibration failed
+    gcs:send_text(0,"Initialisation failed: " .. _dev_name[_nau7802_index])
+    gcs:send_text(0,"NAU7802 calibration status: " .. tostring(cal_status))
+    return
+  end
+
+  return init_nau7802, 100
+end
+------------------------------------------------------------------------
+
+
+------------------------------------------------------------------------
+function init()
+
+  -- Init load cells
+  if not(_dev_initialised[THRUST]) then
+      set_device(i2c_thrust, THRUST)
+      return init_nau7802, 100
+  end
+
+  if not(_dev_initialised[TORQUE]) then
+      set_device(i2c_torque, TORQUE)
+      return init_nau7802, 100
+  end
+
+  -- Setup file to record data to
+  -- Scan through directory to find a unique file name.  Break up search over
+  -- with callbacks every 20 file names
+  if not(_file_index_found) then
+    for i = 1,20 do
+      if not (is_file(string.format(_file_name_fmt, _file_index+i))) then
+        -- Set file index to
+        _file_index = _file_index + i
+        _file_index_found = true
+        break
+      end
+
+      -- Reschedule init to continue searching for a file name
+      if i >= 20 then
+        _file_index = _file_index+20
+        return init, 100
+      end
+    end
+  end
+
+  -- local file = assert(io.open(file_name, "w"),"Could not make file: " .. file_name)
+  local file = assert(io.open(string.format(_file_name_fmt, _file_index), "w"), "Could not make file" .. string.format(_file_name_fmt, _file_index))
+  local header = 'Time (ms), Throttle (), RC Out (us), Motor Commutations (1/min), Voltage(V), Current (A), Thrust (g), Torque (kg.cm), ESC Volt (V), ESC Current (A), ESC RPM (rpm), ESC Temperature (deg C)\n'
+  file:write(header)
+  file:close()
+
+  -- Set first throttle step
+  set_next_thr_step()
+
+  _calibration_loaded_flag = load_calibration(THRUST)
+  if _calibration_loaded_flag then
+    gcs:send_text(4,"Thrust calibration values loaded ")
+    -- Advance system state, thrust calibration not needed
+    _sys_state = REQ_CAL_TORQUE_ZERO_OFFSET
+  end
+
+  _calibration_loaded_flag = _calibration_loaded_flag and load_calibration(TORQUE)
+  if _calibration_loaded_flag then
+    gcs:send_text(4,"Torque calibration values loaded ")
+    -- Advance system state, no calibration needed
+    _sys_state = DISARMED
+  end
+
+  -- Init neopixles
+  _led_chan = SRV_Channels:find_channel(98)
+  if not _led_chan then
+    gcs:send_text(6, "LEDs: channel not set")
+    return
```

Comment:
If no LED's then we are done setting up, and doing things like setting up motor 1 below don't need to happen? That seems wrong.